02-express-tutorial/final/07-params-query.js

```diff
@@ -56,3 +56,78 @@ app.get('/api/v1/query', (req, res) => {
 app.listen(5000, () => {
   console.log('Server is listening on port 5000....')
 })
+
+
+
+// const express = require('express');
+// const app = express();
+// //importing from data.js file
+// const {products} = require('./data')
+
+// app.get('/',(req, res)=>{
+//     res.send('<h1>Home Page</h1><a href="/api/products">products</a>')
+// })
+
+// app.get('/api/products', (req,res)=>{
+//     //sending back array with specific products 
+//     const newProducts = products.map((product)=>{
+//         const{id, name, image} = product;
+//         return{id,name,image}
```

Comment:
I really like the comments. It is very important. Even for me to check the code, it is so helpful. But importantly, comment in actual work projects are best way to help your team.
BTW, the rean to destructor Products like you did is to selectively send the properties but it sends all products, since you are using Map function. Code is right but the comment is little otherwise. But good work.